### DISCUSSION TITLE
Add control for changing Weekly Programm mode for lights and shutter (Auto mode <--> Manual mode)

### DISCUSSION THREAD

**Discussion by Denis1983** [SukramJ/aiohomematic#1886] [Ideas]
Please add control for changing Weekly Programm mode for lights and shutter.
Now it is only possible for Thermostat.
![image](https://github.com/user-attachments/assets/71c7fd38-b9b4-4052-842f-a67a2ef895ab)

It would be useful for example to disable morning lights/shutter programm (sunset emulation) for a days-off/holidays.
Or build an automation to disable auto mode for some days/holidays.

Homematic interface.
![image](https://github.com/user-attachments/assets/65294f55-22b8-49d1-9c15-641ac2fbbabd)

**Comment by SukramJ** [comment]
Wenn du mir sagst wie welche Parameter zu verwenden sind, dann kann ich das gerne einbauen.

**Comment by Denis1983** [comment]
I am not sure what exactly is needed, but I will try.
Docu
https://www.eq-3.de/Downloads/eq3/download%20bereich/hm_web_ui_doku/HmIP_Device_Documentation.pdf
I also found here something similliar.
https://github.com/openhab/openhab-addons/blob/main/bundles/org.openhab.binding.homematic/src/main/resources/homematic/generated-descriptions.properties
(Look for WEEK_PROGRAM_TARGET_CHANNEL)

To get:
WEEK_PROFILE.CHANNEL_LOCKS

For HmIP-PDT
channelIndex 6
    Auto Mode 2
	Manual Mode 3

For HmIP-BDT and HmIP-BROLL
channelIndex 7
	Auto Mode 0
	Manual Mode 1

To set:
WEEK_PROGRAM_TARGET_CHANNEL_LOCKS=Channels for mode week program (binary)

WEEK_PROGRAM_TARGET_CHANNEL_LOCK|AUTO_MODE_WITHOUT_RESET=week program: Auto without reset
WEEK_PROGRAM_TARGET_CHANNEL_LOCK|AUTO_MODE_WITH_RESET=week program: Auto with reset (reset without function)
WEEK_PROGRAM_TARGET_CHANNEL_LOCK|MANU_MODE=week program: Manually

For HmIP-PDT
WEEK_PROGRAM_TARGET_CHANNEL_LOCKS = 3

For HmIP-BDT and HmIP-BROLL	
WEEK_PROGRAM_TARGET_CHANNEL_LOCKS = 4

This is how the set interface looks like in CCU3:
![image](https://github.com/user-attachments/assets/6f59530b-51c5-4ece-968b-e91444fdfe6a)

**Reply by SukramJ** [reply]
Derzeit gibt es in HAHomematic noch nicht die Basis, um die gewünsche Funktionalität als generischen Ansatz umzusetzen, da aus einem Datenpunkt (WEEK_PROGRAM_CHANNEL_LOCKS) gelesen, und in einen anderen Datenpunkt (WEEK_PROGRAM_TARGET_CHANNEL_LOCK) geschrieben werden muss.

Da das Thema bisher noch niemanden interessiert hat wäre es vorerst das Beste, wenn Du die gewünschte Funktionalität mit einem Input-Select-Helper (feste Werteliste aus WEEK_PROGRAM_TARGET_CHANNEL_LOCK) und einer Automatisierung selber löst. Die Automatisierung ändert dann mit [set_device_value](https://github.com/danielperna84/custom_homematic#homematicip_localset_device_value) den ausgewählten Wert für WEEK_PROGRAM_TARGET_CHANNEL_LOCK.
Den aktuellen Wert kann man wohl aus WEEK_PROGRAM_CHANNEL_LOCKS (Integer) lesen, und damit wiederum den Input-Select-Helper anpassen. WEEK_PROGRAM_CHANNEL_LOCKS kann man per [unignore](https://github.com/danielperna84/custom_homematic#unignore-device-parameters) der Integration hinzufügen.

**Reply by SukramJ** [reply]
Wenn Du das implementiert hast, dann könntest Du die entsprechenden Code-Schnippsel hier im Wiki ablegen.
Wenn ich mich dann mal an eine Implementierung setze, dann hab ich schon mal ein funktionsfähiges Beispiel.

**Reply by Baxxy13** [reply]
Macht Euch das Leben damit nicht so schwer. 
Es geht viel einfacher über den COMBINED_PARAMETER im "Wochenprogramm-Kanal". Zumindest fürs "Setzen". 
Wie der String aussehen muss kann man im Dann eines WebUI-Programmes erfahren. 
Hinweis noch: Der Manu-Modus ist flüchtig, also nach Stromausfall ist erstmal wieder Auto aktiv.

**Reply by Denis1983** [reply]
> Macht Euch das Leben damit nicht so schwer. Es geht viel einfacher über den COMBINED_PARAMETER im "Wochenprogramm-Kanal". Zumindest fürs "Setzen". Wie der String aussehen muss kann man im Dann eines WebUI-Programmes erfahren. Hinweis noch: Der Manu-Modus ist flüchtig, also nach Stromausfall ist erstmal wieder Auto aktiv.

Hi, danke für den Tipp, ich konnte aber in Homematic keine Doku finden, wie man COMBINED_PARAMETER nuzt.

**Reply by Baxxy13** [reply]
> in Homematic keine Doku

Was meinst du mit "in Homematic"? 
Die WebUI nutzt den COMBINED_PARAMETER im Dann von WebUI-Programmen wenn man die Option "Kanalaktion" hat.
Dokumentiert ist das nirgends.

Mittels Homematic(IP) Local von Homeassistant aus kannst du den COMBINED_PARAMETER mit der Aktion: "Homematic(IP) Local: Wert eines Geräteparameters schreiben" beschreiben.
Beispiel:
```yaml
action: homematicip_local.set_device_value
data:
  device_id: b33db650c095c9ea88b1a2658579438e
  parameter: COMBINED_PARAMETER
  value: "WPTCLS=7,WPTCL=0"
  channel: 7
  value_type: string
```
